src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1202,49 +1313,82 @@ CHIP_ERROR CASESession::SendSigma2()
         nocCert = MutableByteSpan{};
     }
 
-    ReturnErrorOnFailure(tlvWriter.PutBytes(TLV::ContextTag(kTag_TBEData_Signature), tbsData2Signature.ConstBytes(),
+    ReturnErrorOnFailure(tlvWriter.PutBytes(TLV::ContextTag(ToRaw(TBEDataTags::kSignature)), tbsData2Signature.ConstBytes(),
                                             static_cast<uint32_t>(tbsData2Signature.Length())));
 
     // Generate a new resumption ID
     ReturnErrorOnFailure(DRBG_get_bytes(mNewResumptionId.data(), mNewResumptionId.size()));
-    ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(kTag_TBEData_ResumptionID), mNewResumptionId));
+    ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(ToRaw(TBEDataTags::kResumptionID)), mNewResumptionId));
 
     ReturnErrorOnFailure(tlvWriter.EndContainer(outerContainerType));
     ReturnErrorOnFailure(tlvWriter.Finalize());
-    msg_r2_signed_enc_len = static_cast<size_t>(tlvWriter.GetLengthWritten());
-
+    msg_r2_signed_enc_len   = static_cast<size_t>(tlvWriter.GetLengthWritten());
+    output.encrypted2Length = msg_r2_signed_enc_len + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES;
```

Comment:
It's too bad the data type of the encrypted message doesn't maintain its length on its own and it has to be ridden-along here.

Reply:
I thought of using `ScopedMemoryBufferWithSize` for `msg_R2_Encrypted` but it gives the actual allocated size, and we are instead using the `LengthWritten` by the TLVWriter. Or should I add resize function to it?

Reply:
Not for now.